app-editors/vscode/vscode-1.56.2.ebuild

```diff
@@ -0,0 +1,105 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit desktop pax-utils xdg
+
+DESCRIPTION="Multiplatform Visual Studio Code from Microsoft"
+HOMEPAGE="https://code.visualstudio.com"
+SRC_URI="
+	amd64? ( https://update.code.visualstudio.com/${PV}/linux-x64/stable -> ${P}-amd64.tar.gz )
+	arm? ( https://update.code.visualstudio.com/${PV}/linux-armhf/stable -> ${P}-arm.tar.gz )
+	arm64? ( https://update.code.visualstudio.com/${PV}/linux-arm64/stable -> ${P}-arm64.tar.gz )
+"
+
+RESTRICT="mirror strip bindist"
+
+LICENSE="
+	Apache-2.0
+	BSD
+	BSD-1
+	BSD-2
+	BSD-4
+	CC-BY-4.0
+	ISC
+	LGPL-2.1+
+	Microsoft-vscode
+	MIT
+	MPL-2.0
+	openssl
+	PYTHON
+	TextMate-bundle
+	Unlicense
+	UoI-NCSA
+	W3C
+"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~arm ~arm64"
+IUSE=""
+
+RDEPEND="
+	app-accessibility/at-spi2-atk
+	app-crypt/libsecret[crypt]
+	dev-libs/nss
+	media-libs/alsa-lib
+	media-libs/libpng:0/16
+	x11-libs/cairo
+	x11-libs/gtk+:3
+	x11-libs/libnotify
+	x11-libs/libxkbcommon
+	x11-libs/libxkbfile
+	x11-libs/libXScrnSaver
+	x11-libs/libXtst
+	x11-libs/pango
+"
+
+QA_PREBUILT="
+	/opt/vscode/code
+	/opt/vscode/libEGL.so
+	/opt/vscode/libffmpeg.so
+	/opt/vscode/libGLESv2.so
+	/opt/vscode/libvulkan.so*
+	/opt/vscode/chrome-sandbox
+	/opt/vscode/libvk_swiftshader.so
+	/opt/vscode/swiftshader/libEGL.so
+	/opt/vscode/swiftshader/libGLESv2.so
+	/opt/vscode/resources/app/extensions/*
+	/opt/vscode/resources/app/node_modules.asar.unpacked/*
+"
+
+pkg_setup() {
+	if use amd64; then
+		S="${WORKDIR}/VSCode-linux-x64"
+	elif use arm; then
+		S="${WORKDIR}/VSCode-linux-armhf"
+	elif use arm64; then
+		S="${WORKDIR}/VSCode-linux-arm64"
+	else
+		die "Visual Studio Code only supports amd64, arm and arm64"
+	fi
+}
```

Comment:
Just for the record, I see 
```
app-editors/vscode
  VariableScope: version 1.56.2: variable 'S' used in 'pkg_setup', lines 73, 75, 77
  VariableScope: version 1.56.2: variable 'WORKDIR' used in 'pkg_setup', lines 73, 75, 77
```
but even after checking PMS I don't think this is a problem for us.

Reply:
WORKDIR is simply not defined in pkg_* phases.

PMS also says: "... variables that are not meaningful in a given phase may be unset or set to any value." While Portage currently doesn't do this, it will do the latter for EAPI 8.

Solution: Move assignment of S to src_unpack.

Reply:
Fixed with 79db57d9916a0e25f4c0486041e9ad8f228dfe66 - thanks @ulm

Reply:
Great! Thank you for your help! I will mirror these changes in the `vscodium` PR.